### ISSUE TITLE
Layer Names tf.keras.applications vs keras.applications not matching

### ISSUE THREAD

**Issue by MiWeiss** [tensorflow/tensorflow#36237]
Not sure if this is a documentation issue or a functional bug. 

## Description of issue:
The layer names in 'tf.keras.applications' are not consistent with the layer names in 'keras.applications'.

## Example:
`keras.applications.resnet50.ResNet50(weights='imagenet').summary()` prints the following layers:

> (...)
> __________________________________________________________________________________________________
> **conv1_pad** (ZeroPadding2D)       (None, 230, 230, 3)  0           input_3[0][0]                    
> __________________________________________________________________________________________________
> **conv1** (Conv2D)                  (None, 112, 112, 64) 9472        conv1_pad[0][0]                  
> __________________________________________________________________________________________________
> **bn_conv1** (BatchNormalization)   (None, 112, 112, 64) 256         conv1[0][0]                      
> __________________________________________________________________________________________________
> **activation_50** (Activation)      (None, 112, 112, 64) 0           bn_conv1[0][0]                   
> __________________________________________________________________________________________________
> **pool1_pad** (ZeroPadding2D)       (None, 114, 114, 64) 0           activation_50[0][0]              
> __________________________________________________________________________________________________
> (...)

                           
`tf.keras.applications.resnet50.ResNet50(weights='imagenet').summary()` prints the following layers:

> (...)
> __________________________________________________________________________________________________
> **conv1_pad** (ZeroPadding2D)       (None, 230, 230, 3)  0           input_6[0][0]                    
> __________________________________________________________________________________________________
> **conv1_conv** (Conv2D)             (None, 112, 112, 64) 9472        conv1_pad[0][0]                  
> __________________________________________________________________________________________________
> **conv1_bn** (BatchNormalization)   (None, 112, 112, 64) 256         conv1_conv[0][0]                 
> __________________________________________________________________________________________________
> **conv1_relu** (Activation)         (None, 112, 112, 64) 0           conv1_bn[0][0]                   
> __________________________________________________________________________________________________
> **pool1_pad** (ZeroPadding2D)       (None, 114, 114, 64) 0           conv1_relu[0][0]                 
> __________________________________________________________________________________________________
> (...)

This can lead to errors if code relying on layer names is migrated from `keras` to `tf.keras`. Even after looking for it for quite a bit, I have not found any documentation explaining the change of layer names. Also, I manually needed to map `keras `layer names to `tf.keras` layer names, which would be avoidable with some nice documentation.

## URL(s) with the issue:
Probably this should be mentioned in the migration docs or the applications docs.
https://www.tensorflow.org/guide/migrate#a_note_on_slim_contriblayers
https://www.tensorflow.org/api_docs/python/tf/keras/applications

## Usage example

The following code snippet runs when applications is imported from `keras `but not with `tf.keras`, as the layer is not found.

```
loaded_model = applications.ResNet50(weights='imagenet')

partial_model = Model(inputs=loaded_model.input, outputs=loaded_model.get_layer('res5c_branch2c').output)
```

Some easily findable documentation should be explaining why the layer is not found (renamed layer names) and where to find the mapping from `keras` layer names to `tf.keras` layer names (i.e., the mapping from `res5c_branch2c ` to `conv5_block3_3_conv`, which is the layer name used in tf.keras).

## Used versions in these examples
tensorflow 2.1.0
Keras 2.3.1

## Final Remark
I'm not sure if renaming the layers was that useful... While the tf.keras layer names might be better readable, papers and tutorials often refer to the keras label names - and users now seem to have to find a mapping themself.

**Comment by CNOCycle** [comment]
Similar issues: #33459, #35336

**Comment by qlzh727** [comment]
Please check https://github.com/keras-team/keras-applications/pull/173

**Comment by qlzh727** [comment]
See https://github.com/tensorflow/tensorflow/issues/33459#issuecomment-592758846. 

In short, keras-team/keras-application was exporting an old version of implementation, and TF is exporting a new version. Both of them should behave correctly, but the implementation detail is little bit different. You should rely on tf.keras.applications for the latest implementation.

**Comment by MiWeiss** [comment]
This may explain the related issue of broken weights. However, it seems no to resolve this issue (which is a documentation issue).

We still have the situation that the Resnet from `tf.keras` has to be handled differently than previous keras versions. There does not seem to be any documentation explaining how to migrate, which I think many will want to do when switching from tf1 to tf2.

Do you agree?

**Comment by MiWeiss** [comment]
@qlzh727 Would you consider reopening this **documentation** issue?

**Comment by qlzh727** [comment]
Sure. Sorry for the late reply.

**Comment by nikitamaia** [comment]
I don't think this is an issue anymore. Please [see this gist here](https://colab.research.google.com/gist/nikitamaia/ef5a0faad77c562d22a717f12622d67f/issue_36237.ipynb), which prints `keras.applications.resnet50.ResNet50(weights='imagenet').summary()`. 
Seem to be the same as `tf.keras` now. Note that the usage example provided will now also fail for `keras` as well as `tf.keras` since the layer names are now consistent.

Closing this issue now since it not longer seems relevant (though please add a note if I missed something).
